hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/cluster/strategy/PartitionAwareClusteringPlanStrategy.java

```diff
@@ -64,6 +65,22 @@ public PartitionAwareClusteringPlanStrategy(HoodieTable table, HoodieEngineConte
    */
   protected Pair<Stream<HoodieClusteringGroup>, Boolean> buildClusteringGroupsForPartition(String partitionPath, List<FileSlice> fileSlices) {
     HoodieWriteConfig writeConfig = getWriteConfig();
+    
```

Comment:
Binary copy related plan strategy should overwrite this buildClusteringGroupsForPartition function which will build groups based on schema.

Reply:
Yeah, this will be changed.